src/vs/base/parts/storage/node/storage.ts

```diff
@@ -34,6 +35,7 @@ export class SQLiteStorageDatabase implements IStorageDatabase {
 
 	get onDidChangeItemsExternal(): Event<IStorageItemsChangeEvent> { return Event.None; } // since we are the only client, there can be no external changes
 
+	private static readonly DEFAULT_MODE = OPEN_READWRITE | OPEN_CREATE; // default mode when connecting to the DB
```

Comment:
I would prefer if we do not use any default mode but leave that up to the SQLite implementation. We only set the mode to readonly when instructed, but otherwise leave it as is.